### PR TITLE
Bindings on steroids

### PR DESCRIPTION
**Pull request by willsb**
## What's this?
Literally #2785 except harder, better, faster and stronger

### Relationships
Supersedes #2785

## Review hints
Please review each commit individually, since it will make it much simpler to understand what's going on

## :squid: Permissions
Me and @Odrakir

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/Reactive/IReactive.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Toggl.Foundation.MvvmCross.Reactive
+{
+    public interface IReactive<out TBase>
```
Comment (kornelijepetak) [inline Toggl.Foundation.MvvmCross/Reactive/IReactive.cs]: Honestly, why not only `T`?

**Comment by kornelijepetak** [comment]
Also, this PR should be merged ASAP to avoid too many merge conflicts during collateral RXfactorings...

**Comment by willsb** [comment]
You can read the reasoning on #2785. Also, this is to make autocompletion better and easier to reason about.

**Comment by kornelijepetak** [comment]
I did read it afterwards. 👍

File: Toggl.Daneel/Extensions/Binders/ReactiveTableViewBinder.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@
 namespace Toggl.Daneel.ViewSources
 {
     public class ReactiveTableViewBinder<TModel, TCell> : IDisposable
-    where TCell : BaseTableViewCell<TModel>
+        where TCell : BaseTableViewCell<TModel>
```
Comment (Odrakir) [inline Toggl.Daneel/Extensions/Binders/ReactiveTableViewBinder.cs]: 👀

**Comment by willsb** [comment]
Not only that, but it also adds the changes to Android :)

**Pull request merged**